api/resource/resource.go

```diff
@@ -296,20 +295,37 @@ func (r *Resource) MustYaml() string {
 	return string(yml)
 }
 
+func (r *Resource) getOptions() *types.GenArgs {
+	annotations := r.GetAnnotations()
+	if genOptsAnno, ok := annotations[utils.BuildAnnotationsGenOptions]; ok {
+		var genOpts types.GeneratorArgs
+		yaml.Unmarshal([]byte(genOptsAnno), &genOpts)
+		return types.NewGenArgs(&genOpts)
+	}
+	return &types.GenArgs{}
```

Comment:
Should this return `nil`? There are a lot of places that seem to interpret nil as "no options configured", presumably because without an accessor nothing necessarily initialized the struct.

Reply:
done